camel/agents/chat_agent.py

```diff
@@ -333,6 +324,16 @@ def step(
                 called_funcs.append(func_record)
             else:
                 # Function calling disabled or chat stopped
+
+                # Loop over responses terminators
+                for terminator in self.response_terminators:
+                    self.terminated, termination_reason = \
```

Comment:
I do not like this overwriting the reasons in the loop. You need to put the results into a list, reduce it after the for loop and mutate self.terminated only once